adafruit_bno055.py

```diff
@@ -108,7 +110,13 @@ class BNO055:
     def __init__(self, i2c, address=0x28):
         self.i2c_device = I2CDevice(i2c, address)
         self.buffer = bytearray(2)
-        self.init()
+        chip_id = self._read_register(_ID_REGISTER)
+        if chip_id != _CHIP_ID:
+            raise RuntimeError("bad chip id (%x != %x)" % (chip_id, _CHIP_ID))
+        self.reset()
+        self._write_register(_POWER_REGISTER, _POWER_NORMAL)
+        self._write_register(_PAGE_REGISTER, 0x00)
+        self._write_register(_TRIGGER_REGISTER, 0x00)
```

Comment:
The setting of mode and the delay after it are necessary for this sensor to work. Removing them breaks it.
You need to add this at the end:
```python
        self.mode = NDOF_MODE
        time.sleep(0.2)
```